### ISSUE TITLE
auto_route does not always navigate to route in nested router

### ISSUE THREAD

**Issue by KCL-TelferAA** [Milad-Akarie/auto_route_library#852]
Hi @Milad-Akarie,

Thank you for `auto_route`! It's an awesome tool that makes navigation a joy in Flutter :)

Unfortunately we've been experiencing a very serious issue for a couple of days and we cannot seem to figure out what is wrong. 

We have the following route setup:
```dart
AutoRoute(
      path: '',
      initial: true,
      page: MainAutoTabsPage,
      guards: [AuthGuard],
      children: [
        AutoRoute(
          page: EmptyRouterPage,
          name: 'DashboardRouter',
          children: [
            AutoRoute(initial: true, page: DashboardPage),
          ],
        ),
        AutoRoute(
          page: EmptyRouterPage,
          name: 'ShowsRouter',
          children: [
            AutoRoute(initial: true, page: ShowsListingPage),
            AutoRoute(page: ShowDetailsPage),
          ],
        ),
      ],
    )
...
```

We are using `AutoTabsScaffold` for navigation like this:
```dart
Widget build(context) {
    return Theme(
      data: commonTheme,
      child: AutoTabsScaffold(
      routes: [
        DashboardRouter(),
        ShowsRouter(),
        MoreRoute(),
      ],
      bottomNavigationBuilder: (BuildContext context, TabsRouter tabsRouter) =>
          DashboardNavBar(tabsRouter, context),
    ));
  }
```

In the `DashboardPage` we navigate to `ShowDetailsPage` like this:
```dart
...
TextButton(
                child: Text('Continue Watching ${lastWatchedShow.title}'),
onPressed: () {
                  context.navigateTo(
                    MainAutoTabsRoute(
                      children: [
                        ShowsRouter(
                          children: [
                            // Must push child routes in order to get back button
                            ShowsListingRoute(),
                            ShowDetailsRoute(
                              show: lastWatchedShow,
                            )
                          ],
                        ),
                      ],
                    ),
                  );
                },
               ),
...
```

The problem we are having is on first click of "Continue Watching..." we sometimes get the `ShowDetailsRoute` route to display, but the vast majority of the time, the screen refreshes and we get `DashboardRoute` instead. The behaviour is random.

Do you think this might be a timing or async issue?

**Comment by Milad-Akarie** [comment]
@KCL-TelferAA that's really weird. I can't think of any reason why this should happen. 
where are you pushing from?

**Comment by KCL-TelferAA** [comment]
> @KCL-TelferAA that's really weird. I can't think of any reason why this should happen.
> where are you pushing from?

I'm pushing from DashboardRoute route.

This is where we are:
`MainAutoTabsRoute -> DashboardRouter -> DashboardRoute` 

We want to go here:
`MainAutoTabsRoute -> ShowsRouter -> ShowDetailsRoute` 

This does not happen when we navigate within a nested route, only when we navigate across nested routes.

We'll look at an older version of the plugin to see if that will help us move forward.

We'll see if we can setup a repo that reproduces the behaviour.

**Comment by Milad-Akarie** [comment]
@KCL-TelferAA That would be helpful, because I recreated the same setup and it navigated as expected everytime.

**Comment by KCL-TelferAA** [comment]
@Milad-Akarie Ok. Maybe in stripping out a few things for the sample repo we'll figure out what is happening.

**Comment by KCL-TelferAA** [comment]
Hi @Milad-Akarie, we figured out what the issue was while creating the sample repo.

Our `AuthGuard` was making unnecessary calls to the authentication web service every time the guard was called. We didn't delve very deeply into the issue, but it seems the delay created by the HTTP request caused something to not work properly while navigating from `MainAutoTabsRoute -> DashboardRouter -> DashboardRoute` to `MainAutoTabsRoute -> ShowsRouter -> ShowDetailsRoute`. 

We did note that `resolver.next(true);` was called in the guard after the HTTP request completed, but it still ended up navigating to the root route `MainAutoTabsRoute` only--not the child routes. After removing the HTTP request everything started functioning as expected.

Thanks for taking the time to answer our question. This is a great tool you have here. Let us know how we can support you in it.
